segmentation violation while deploying an azurerm_linux_function_app

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.11.4

### AzureRM Provider Version

4.27.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_function_app" "onevision-billing-function-app" {
  name                = "some-name"
  resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location

  storage_account_name       = azurerm_storage_account.my-storage.name
  storage_account_access_key = azurerm_storage_account.my-storage.primary_access_key
  service_plan_id            = azurerm_service_plan.my-billing-plan.id

  public_network_access_enabled = true
  virtual_network_subnet_id     = azurerm_subnet.subnet.id

  site_config {
    always_on = true
    application_stack {
      python_version = "3.12"
    }
  }

  app_settings = {
    "WEBSITE_RUN_FROM_PACKAGE"              = "1"
    "APPINSIGHTS_INSTRUMENTATIONKEY"        = azurerm_application_insights.my-app-insights.instrumentation_key
    "APPLICATIONINSIGHTS_CONNECTION_STRING" = azurerm_application_insights.my-app-insights.connection_string
  }
}
```

### Debug Output/Panic Output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4d04b4f]

goroutine 300 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice.(*LinuxFunctionAppResource).Read.LinuxFunctionAppResource.Read.func1({0x89bffb0, 0xc000ed71f0}, {0xc00233b208, {0x89c3938, 0xc000283e60}, 0xc0037a3e00, 0x0, {0x89c41b8, 0xdba6b00}})
  github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/linux_function_app_resource.go:744 +0xbef
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func3({0x89bffb0, 0xc000ed71f0}, 0x89a6cb0?, {0x7113bc0?, 0xc00233b208?})
  github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:69 +0x145
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.(*ResourceWrapper).diagnosticsWrapper.diagnosticsWrapper.func11({0x89bffb0?, 0xc000ed71f0?}, 0x45d964b800?, {0x7113bc0?, 0xc00233b208?})
  github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:188 +0x59
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000111d00, {0x89bff08, 0xc003a7aba0}, 0xc0037a3e00, {0x7113bc0, 0xc00233b208})
  github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:855 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000111d00, {0x89bff08, 0xc003a7aba0}, 0xc003f12270, {0x7113bc0, 0xc00233b208})
  github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:1149 +0x53a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000a0f368, {0x89bff08?, 0xc003a7aae0?}, 0xc003a6a380)
  github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:739 +0x6c5
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xc00085cd80, {0x89bff08?, 0xc003a7a810?}, 0xc003a6a380)
  github.com/hashicorp/terraform-plugin-mux@v0.17.0/tf5muxserver/mux_server_ReadResource.go:35 +0x190
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00084e3c0, {0x89bff08?, 0xc003a7a030?}, 0xc000ed6cb0)
  github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:784 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x7fbdc00, 0xc00084e3c0}, {0x89bff08, 0xc003a7a030}, 0xc0037a3900, 0x0)
  github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:575 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001c84000, {0x89bff08, 0xc003a55f50}, 0xc003931e00, 0xc001a5fbf0, 0xdb73088, 0x0)
  google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc001c84000, {0x89c3ea8, 0xc001c8a1a0}, 0xc003931e00)
  google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
  google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 23
  google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-azurerm_v4.27.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

An RBAC role / permission is missing in my case, and usually the azurerm provider tells which permission is missing. I am expecting to have a message telling which permission is missing, rather than this panic output.

### Actual Behaviour

Panic output, very likely because the service principal running the terraform deployment is missing a permission. When provided with Owner role, the deployment succeeds. When provided with the bare minimum permissions, it throws this panic output. I'd be grateful if someone could tell which permissions are required. Currently, it has at least the following permissions:

    "Microsoft.Insights/components/read",
    "Microsoft.Insights/components/write",
    "Microsoft.Insights/components/delete",
     "Microsoft.OperationalInsights/workspaces/read",
    "Microsoft.OperationalInsights/workspaces/sharedkeys/read",
    "Microsoft.Resources/subscriptions/resourcegroups/read",
    "Microsoft.Resources/subscriptions/resourcegroups/write",
    "Microsoft.Resources/subscriptions/resourcegroups/delete",
    "Microsoft.Web/serverFarms/read",
    "Microsoft.Web/serverFarms/write",
    "Microsoft.Web/serverFarms/delete",
    "Microsoft.Web/sites/read",
    "Microsoft.Web/sites/write",
    "Microsoft.Web/sites/delete",
    "Microsoft.Web/sites/config/write",
    "Microsoft.Web/sites/config/read",
    "Microsoft.Web/sites/config/delete",
    "Microsoft.Web/sites/config/list/action",
 
It is also a Storage Blob Data Contributor and a Storage Account Contributor.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appservice/linux_function_app_resource.go
+++ b/internal/services/appservice/linux_function_app_resource.go
@@ -748,14 +748,14 @@ func (r LinuxFunctionAppResource) Read() sdk.ResourceFunc {
 			}
 
 			basicAuthFTP := true
-			if basicAuthFTPResp, err := client.GetFtpAllowed(ctx, *id); err != nil && basicAuthFTPResp.Model != nil {
+			if basicAuthFTPResp, err := client.GetFtpAllowed(ctx, *id); err != nil || basicAuthFTPResp.Model == nil {
 				return fmt.Errorf("retrieving state of FTP Basic Auth for %s: %+v", id, err)
 			} else if csmProps := basicAuthFTPResp.Model.Properties; csmProps != nil {
 				basicAuthFTP = csmProps.Allow
 			}
 
 			basicAuthWebDeploy := true
-			if basicAuthWebDeployResp, err := client.GetScmAllowed(ctx, *id); err != nil && basicAuthWebDeployResp.Model != nil {
+			if basicAuthWebDeployResp, err := client.GetScmAllowed(ctx, *id); err != nil || basicAuthWebDeployResp.Model == nil {
 				return fmt.Errorf("retrieving state of WebDeploy Basic Auth for %s: %+v", id, err)
 			} else if csmProps := basicAuthWebDeployResp.Model.Properties; csmProps != nil {
 				basicAuthWebDeploy = csmProps.Allow

